dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java

```diff
@@ -0,0 +1,158 @@
+        athenaConnectionParam.setDatabase(athenaParam.getDatabase());
+        athenaConnectionParam.setDriverClassName(this.getDatasourceDriver());
+        athenaConnectionParam.setValidationQuery(this.getValidationQuery());
+        athenaConnectionParam.setProps(athenaParam.getOther());
+
+        return athenaConnectionParam;
+    }
+
+    @Override
+    public ConnectionParam createConnectionParams(String connectionJson) {
+        return JSONUtils.parseObject(connectionJson, AthenaConnectionParam.class);
+    }
+
+    @Override
+    public String getDatasourceDriver() {
+        return Constants.COM_ATHENA_JDBC_DRIVER;
+    }
+
+    @Override
+    public String getValidationQuery() {
+        return Constants.ATHENA_VALIDATION_QUERY;
+    }
+
+    @Override
+    public String getJdbcUrl(ConnectionParam connectionParam) {
+        AthenaConnectionParam
+            athenaConnectionParam = (AthenaConnectionParam) connectionParam;
+        if (!StringUtils.isEmpty(athenaConnectionParam.getOther())) {
+            return String.format("%s%s", athenaConnectionParam.getJdbcUrl(), athenaConnectionParam.getOther());
+        }
+        return athenaConnectionParam.getJdbcUrl();
+    }
+
+    @Override
+    public Connection getConnection(ConnectionParam connectionParam) throws ClassNotFoundException, SQLException {
+        AthenaConnectionParam athenaConnectionParam = (AthenaConnectionParam) connectionParam;
+        Class.forName(this.getDatasourceDriver());
+        return DriverManager.getConnection(this.getJdbcUrl(connectionParam),
+            athenaConnectionParam.getUser(), PasswordUtils.decodePassword(athenaConnectionParam.getPassword()));
+    }
+
+    @Override
+    public DbType getDbType() {
+        return DbType.ATHENA;
+    }
+
+    @Override
+    public DataSourceProcessor create() {
+        return new AthenaDataSourceProcessor();
+    }
+
+    private String transformOther(Map<String, String> otherMap) {
+        if (MapUtils.isNotEmpty(otherMap)) {
+            List<String> list = new ArrayList<>(otherMap.size());
+            otherMap.forEach((key, value) -> list.add(String.format("%s=%s", key, value)));
+            return String.join(Constants.SEMICOLON, list);
+        }
+        return null;
+    }
+
+    private Map<String, String> parseOther(String other) {
+        Map<String, String> otherMap = new LinkedHashMap<>();
+        if (StringUtils.isEmpty(other)) {
+            return otherMap;
+        }
+        String[] configs = other.split(Constants.SEMICOLON);
+        for (String config : configs) {
+            otherMap.put(config.split(Constants.EQUAL_SIGN)[0], config.split(Constants.EQUAL_SIGN)[1]);
+        }
+        return otherMap;
+    }
+
+    @Override
+    protected void checkHost(String host) {
+        // Do not need to set the host, nothing to do
+    }
+
+    @Override
+    protected void checkDatabasePatter(String database) {
+        // Do not need to set the database, nothing to do
+    }
```

Comment:
we can remove those method

Reply:
if remove those method, will cause the parent class parameter check error, `org.apache.dolphinscheduler.plugin.datasource.api.datasource.AbstractDataSourceProcessor#checkDatasourceParam` , I find  POSTGRESQL  are also problems with database verification

Reply:
👌